### PR TITLE
Bugs/mvn3.3.3

### PR DESCRIPTION
**Pull request by rickalm**
## What changes were proposed in this pull request?

Changing the URI for Maven 3.3.3, version is no longer available on previous endpoint
## How was this patch tested?

Successfully built spark using make_distribution after the patch was applied

### CODE REVIEW

**Comment by holdenk** [comment]
cc @srowen for build update

File: build/mvn [hunk 1]
```diff
@@ -72,7 +72,7 @@ install_mvn() {
   local MVN_VERSION="3.3.3"
 
   install_app \
-    "https://www.apache.org/dyn/closer.lua?action=download&filename=/maven/maven-3/${MVN_VERSION}/binaries" \
```
Comment (holdenk) [inline build/mvn]: Interestingly enough 3.3.9 works through the URL but 3.3.3 doesn't appear to anymore - I'm not sure if maybe just one of the mirrors is screwed up or if this is a more intentional change.

**Comment by rickalm** [comment]
Was considering withdrawing the PR. Here are my thoughts

Tag v1.6.2 specifies mvn 3.3.3 (unless user intentionally overrides). Users who previously built wanting to replicate their previous build have two choices:

#1, Apply this patch allowing v1.6.2 to be build as previous
#2, Apply PR #14066 bumping maven to 3.3.9 (btw this PR misses updating the pom.xml

The purist in me thinks #1 is the correct approach, but I'll defer to general wisdom

@srowen thoughts ?

**Comment by srowen** [comment]
This is already in `branch-1.6`: https://github.com/apache/spark/blob/branch-1.6/build/mvn
We can't change releases/tags of course. Yeah, this is because non-current versions eventually get moved off active mirrors. It's not great but only a problem if you don't actually have Maven installed. I'd expect most people have 3.3.x available locally anyway.

I think we don't want to download every time from dist, or that's discouraged.

**Pull request closed without merging**